pyro/infer/autoguide/guides.py

```diff
@@ -862,6 +862,7 @@ class AutoMultivariateNormal(AutoContinuous):
         (unconstrained transformed) latent variable.
     """
 
+    scale_constraint = constraints.softplus_positive
     scale_tril_constraint = constraints.softplus_lower_cholesky
```

Comment:
i'm confused. it looks like we rename torch's `lower_cholesky to `softplus_lower_cholesky` even though the former uses a an `exp` transform for the diagonal?  or am i missing something?

Reply:
[softplus_lower_cholesky](https://github.com/pyro-ppl/pyro/blob/07193b8a21eb5ff4c117097fed470024c6a794e1/pyro/distributions/constraints.py#L110) is @vitkl's more numerically stable constraint, which dispatches to a [SoftplusLowerCholeskyTransform](https://github.com/pyro-ppl/pyro/blob/07193b8a21eb5ff4c117097fed470024c6a794e1/pyro/distributions/transforms/softplus.py#L36) which uses a `torch.nn.functional.softplus()` under the hood. @vitkl found this to be more numerically stable after extensive experiments. See #2823 and [supporting experiments](https://github.com/pyro-ppl/sandbox/tree/master/2021-03-softplus_scales).